### ISSUE TITLE
[rabbitmq-ha] How to use Secure Websockets

### ISSUE THREAD

**Issue by AndrewBedscastle** [helm/charts#22042]
(k8s beginner, experienced docker user)

After searching for a long time I did not find an answer to the following problem:

We want to upgrade the normal web sockets to secure websockets

The values.yaml lets me specify the corresponding .pem files, but how can I reference them?
Do I need a volume for that? 

`rabbitmqWebMQTTPlugin:
  enabled: false

  ## Web MQTT configuration:
  config: |
    # web_mqtt.ssl.port       = 12345
    # web_mqtt.ssl.backlog    = 1024
    # web_mqtt.ssl.certfile   = /etc/cert/cacert.pem
    # web_mqtt.ssl.keyfile    = /etc/cert/cert.pem
    # web_mqtt.ssl.cacertfile = /etc/cert/key.pem
    # web_mqtt.ssl.password   = changeme`

Are these files referenced above created after specifying the following option?

`rabbitmqCert:
  enabled: false

  ## Create a new secret using these values
  cacertfile: |   <--- value here
  certfile: | <---- value here
  keyfile: |` <---- value here

If I understand that correctly, it is possible to provide base64 encoded certificate data here, see above.

Help appreciated, thank you very much!
I really didn't find any documentation that helped me

**Comment by AndrewBedscastle** [comment]
Meanwhile I managed to create a certificate secret, but I just don't know how to get it running with rabbitmq-ha as it's not documented

Tried:

- rabbitmqCert: enabled true and specified the existing secret (in the same namespace, I checked that)
That gives me: Did not find secret xxx in the container cert

- Uncommented these lines
# web_mqtt.ssl.port = 12345
# web_mqtt.ssl.backlog = 1024
# web_mqtt.ssl.certfile = /etc/cert/cacert.pem
# web_mqtt.ssl.keyfile = /etc/cert/cert.pem
# web_mqtt.ssl.cacertfile = /etc/cert/key.pem
# web_mqtt.ssl.password = changeme`

The port is not opened on the load balancer; the ssl.certfiles / keyfile / cacertfile are reported to be not found, the pods crash

Please guide me in the correct direction. I would help myself but the missing documentation gives me a hard time.

To sum it up:
I want to enable SSL / Secure Websockets (wss) using an existing secret (ingress would be possible as well)

Thank you very much!

**Comment by AndrewBedscastle** [comment]
This can be closed.
I solved it by letting the connection itself remain unsecure and securing it with an nginx ingress with terminates TLS:

See here: https://gist.github.com/jsdevtom/7045c03c021ce46b08cb3f41db0d76da#file-ingress-service-yaml
